### PR TITLE
Bugfix/daemon child dies when worker dies

### PR DESCRIPTION
**Pull request by roengstrom**

### CODE REVIEW

**Comment by roengstrom** [comment]
This is a re-revert of the pull request merged a few days ago followed with a fix for the bug it introduced.

File: src/naemon/utils.c [hunk 1]
```diff
@@ -556,23 +555,13 @@ int daemon_init(void)
...
-	}
-
 	if ((pid = (int)fork()) < 0) {
 		nm_log(NSLOG_RUNTIME_ERROR, "Unable to fork out the daemon process: %s",
 			strerror(errno));
 		return (ERROR);
 	} else if (pid != 0) {
 		/* parent stops - when child is done, we exit here */
-		int return_code = OK;
+		int return_code = EXIT_FAILURE;
```
Comment (tvestelind) [inline src/naemon/utils.c]: Could we just have a comment where this variable is set to `OK`, just to make things a bit clearer.

Comment (roengstrom) [inline src/naemon/utils.c]: Absolutely, fixed!

**Comment by tvestelind** [comment]
I'm OK with merging internally :+1:

**Pull request merged**